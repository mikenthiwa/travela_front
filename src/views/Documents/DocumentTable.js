import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DocumentItem from './DocumentItem';
import './Documents.scss';

class DocumentTable extends Component {
  renderTableHead() {
    return (
      <tr>
        <th className="mdl-data-table__cell--non-numeric bb-md-0 table__head freeze freeze-head ">
          Name
        </th>
        <th className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left">
          Date Added
        </th>
        <th className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left">
          Last Modified
        </th>
        <th className="mdl-data-table__cell--non-numeric table__head table__head--last" />
      </tr>
    );
  }

  render() {
    const {
      documents, menuOpen, toggleMenu,
      openModal, editDocument, setItemToDelete
    } = this.props;
    return (
      <table className="mdl-data-table mdl-js-data-table table__requests">
        <thead>
          {this.renderTableHead()}
        </thead>
        <tbody className="table__body">
          {
            documents.map(document => (
              <DocumentItem
                key={document.id}
                document={document}
                menuOpen={menuOpen}
                toggleMenu={toggleMenu}
                openModal={openModal}
                setItemToDelete={setItemToDelete}
                editDocument={editDocument}
              />
            ))}
        </tbody>
      </table>
    );
  }
}

const documentTablePropTypes = {
  documents: PropTypes.array.isRequired,
  openModal: PropTypes.func.isRequired,
  setItemToDelete: PropTypes.func.isRequired,
  toggleMenu: PropTypes.func.isRequired,
  editDocument: PropTypes.func.isRequired,
  menuOpen: PropTypes.object.isRequired,
};

DocumentTable.propTypes = { ...documentTablePropTypes };

export default DocumentTable;
