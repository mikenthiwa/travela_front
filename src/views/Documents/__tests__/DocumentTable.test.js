import React from 'react';
import { shallow } from 'enzyme';
import DocumentTable from '../DocumentTable';

describe('<DocumentTable />', () => {
  const props = {
    documents: [
      {
        id: '1',
        name: 'Passport',
        cloudinary_public_id: 'e93h236FvT',
        cloudinary_url: 'https://image.jpg',
        userId: '-LMgZQKq6MXAj_41iRWi',
        createdAt: '2018-08-16 12:11:52.181+01',
        updatedAt: '2018-08-16 12:11:52.181+01',
      },
      {
        id: '2',
        name: 'Visa',
        cloudinary_public_id: 'e93h236FvT',
        cloudinary_url: 'https://image.url',
        userId: '-LMgZQKq6MXAj_41iRWi',
        createdAt: '2018-08-16 12:11:52.181+01',
        updatedAt: '2018-08-16 12:11:52.181+01',
      }
    ],
    menuOpen: {
      open: true
    },
    toggleMenu: jest.fn(),
    openModal: jest.fn(),
    setItemToDelete: jest.fn(),
    editDocument: jest.fn()
  };
  it('should render correctly with documents', () => {
    const wrapper = shallow(<DocumentTable {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.table__body').length).toBe(1);
  });

  it('should render DocumentItem', () => {

    const wrapper = shallow(<DocumentTable {...props} />);
    expect(wrapper.find('DocumentItem').length).toBe(2);
  });
});
