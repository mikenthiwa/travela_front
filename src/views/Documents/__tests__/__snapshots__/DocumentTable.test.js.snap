// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentTable /> should render correctly with documents 1`] = `
<table
  className="mdl-data-table mdl-js-data-table table__requests"
>
  <thead>
    <tr>
      <th
        className="mdl-data-table__cell--non-numeric bb-md-0 table__head freeze freeze-head "
      >
        Name
      </th>
      <th
        className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
      >
        Date Added
      </th>
      <th
        className="mdl-data-table__cell--non-numeric table__head pl-sm-100d description-left"
      >
        Last Modified
      </th>
      <th
        className="mdl-data-table__cell--non-numeric table__head table__head--last"
      />
    </tr>
  </thead>
  <tbody
    className="table__body"
  >
    <DocumentItem
      document={
        Object {
          "cloudinary_public_id": "e93h236FvT",
          "cloudinary_url": "https://image.jpg",
          "createdAt": "2018-08-16 12:11:52.181+01",
          "id": "1",
          "name": "Passport",
          "updatedAt": "2018-08-16 12:11:52.181+01",
          "userId": "-LMgZQKq6MXAj_41iRWi",
        }
      }
      editDocument={[Function]}
      key="1"
      menuOpen={
        Object {
          "open": true,
        }
      }
      openModal={[Function]}
      setItemToDelete={[Function]}
      toggleMenu={[Function]}
    />
    <DocumentItem
      document={
        Object {
          "cloudinary_public_id": "e93h236FvT",
          "cloudinary_url": "https://image.url",
          "createdAt": "2018-08-16 12:11:52.181+01",
          "id": "2",
          "name": "Visa",
          "updatedAt": "2018-08-16 12:11:52.181+01",
          "userId": "-LMgZQKq6MXAj_41iRWi",
        }
      }
      editDocument={[Function]}
      key="2"
      menuOpen={
        Object {
          "open": true,
        }
      }
      openModal={[Function]}
      setItemToDelete={[Function]}
      toggleMenu={[Function]}
    />
  </tbody>
</table>
`;
